messages/src/lib.rs

```diff
@@ -187,25 +187,50 @@ pub enum HwError {
     )]
     ControlPortReadFailed,
 
-    /// Failed to read the `EN` register
-    #[cfg_attr(any(test, feature = "std"), error("Failed to read `EN` register"))]
-    EnableReadFailed,
+    /// Failed while adjusting the power mode
+    #[cfg_attr(
+        any(test, feature = "std"),
+        error("Failed while adjusting the power mode")
+    )]
+    PowerModeChangeFailed,
```

Comment:
I was confused by this, specifically how it's different from `PowerEnableWriteFailed`. But I looked at the related PR on the Hubris side, and I think I get it.

Just checking my understanding. When the `SetPowerMode` message is sent, the SP does a sequence of a few actions. For example, if the mode is `PowerMode::Low`, it seems to: enable the efuse; assert `LPMode`; and de-assert `ResetL`.

If that's correct, then I think we should remove that entire message and related code in Hubris. The main reason is that we're giving the host direct control over the `LPMode` and `ResetL` pins specifically because we don't want Hubris to have to worry about that. For example, that whole operation is kind of lying if the module has software override of power control. The request to Hubris will succeed, but the module may not change its power mode at all.

It seems we'll instead want to put that kind of operation in the host `Controller`. The alternative is putting the ability to check for software override of power control into Hubris, which I don't believe we're interested in.

Reply:
Your understanding is correct. I had both methods of control there since I knew in one sense the host wanted control of the individual signals and in another it may not. I am a fan of handling that abstraction at a layer higher than this!

Reply:
Sweet! I think let's snip all that out now, so we don't have the vestigial and hard-to-use `SetPowerMode` message at all. I can follow up implementing the new `Controller::set_power` function described in https://github.com/oxidecomputer/transceiver-control/issues/41.